src/mon/OSDMonitor.cc

```diff
@@ -678,21 +671,36 @@ class OSDUtilizationDumper : public CrushTreeDumper::Dumper<F> {
     return sum > 0 ? sqrt(stddev / sum) : 0;
   }
 
-  bool get_bucket_utilization(int id, int64_t& kb, int64_t& kb_used,
-			      int64_t& kb_avail) const {
-    if (id >= 0) {
-      typedef ceph::unordered_map<int32_t,osd_stat_t> OsdStat;
-
-      OsdStat::const_iterator p = pgm->osd_stat.find(id);
+  double average_utilization() {
+    int64_t kb = 0, kb_used = 0;
+    for (int i = 0; i <= osdmap->get_max_osd(); i++) {
+      if (!osdmap->exists(i) || osdmap->get_weight(i) == 0)
+	continue;
+      int64_t kb_i, kb_used_i, kb_avail_i;
+      if (get_osd_utilization(i, kb_i, kb_used_i, kb_avail_i)) {
+	kb += kb_i;
+	kb_used += kb_used_i;
+      }
+    }
+    return kb > 0 ? 100.0 * (double)kb_used / (double)kb : 0;
+  }
 
-      if (p == pgm->osd_stat.end())
-	return false;
+  bool get_osd_utilization(int id, int64_t& kb, int64_t& kb_used,
+			   int64_t& kb_avail) const {
```

Comment:
the preferred convention is to use pointers and not references for output arguments